File: app/src/main/java/fr/nihilus/music/database/Converters.java
/*
 * Copyright 2017 Thibault Seisel
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package fr.nihilus.music.database;

import android.arch.persistence.room.TypeConverter;
import android.net.Uri;
import android.support.annotation.Nullable;

import java.util.Date;

final class Converters {

    @TypeConverter
    public Date fromTimestamp(@Nullable Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public Long dateToTimestamp(@Nullable Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public Uri fromString(@Nullable String str) {
        // TODO Might return Uri.EMPTY instead of null
        return str == null ? null : Uri.parse(str);
    }

    @TypeConverter
    public String toUriString(@Nullable Uri uri) {
        return uri == null ? null : uri.toString();
    }

}

File: app/src/main/java/fr/nihilus/music/database/Converters.kt
package fr.nihilus.music.database

import android.arch.persistence.room.TypeConverter
import android.net.Uri
import java.util.*

internal class Converters {

    @TypeConverter
    fun fromTimestamp(value: Long?): Date? = value?.let { Date(it) }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? = date?.time

    @TypeConverter
    fun fromString(str: String?): Uri = str?.let { Uri.parse(it) } ?: Uri.EMPTY

    @TypeConverter
    fun toUriString(uri: Uri?): String? = uri?.toString()

}
